src/core/server/workspaces/workspaces_service.ts

```diff
@@ -84,6 +94,7 @@ export class WorkspacesService
 
     return {
       client: this.client as IWorkspaceDBImpl,
+      permissionControl: this.permissionControl as WorkspacePermissionControl,
```

Comment:
Q: Could you give an example of how will we use `permissionControl`?

Reply:
As my understanding. The `permissionControl` will be a result of `setup` method. After `setup` called, we will pass them to anywhere that dependencies on workspace `permissionControl`.